src/wmnut.c

```diff
@@ -131,23 +148,25 @@ void get_ups_info(void)
 {
 	char	value[SMALLBUF];
 	int	retVal;
+	size_t	numa;
 
 	/* Get UPS status */
 	if (get_ups_var ("ups.status", value) > NOK) {
-
-		if (!strncmp(value, "OL", 2))
-			CurHost->ups_status = UPS_ONLINE;
-		else if (!strncmp(value, "OB", 2))
-			CurHost->ups_status = UPS_ONBATT;
-		else if (!strstr(value, "LB"))
-			CurHost->ups_status = UPS_LOWBATT;
-		else if (!strncmp(value, "OVER", 4))
-			CurHost->ups_status = UPS_OVERLOAD;
-		else
-		CurHost->ups_status = UPS_DEFAULT;
+		if (strstr(value, "OL")) {
```

Comment:
Currently we are lucky that no keyword in the table we care about is a sub-string of another. This was not really a consideration for NUT standard (`nut-names.txt`) and technically drivers or NUT forks can make up their own words although it is frowned upon.

Generally I'd be in favor of porting https://github.com/networkupstools/nut/blob/8becada220ceb8e2634c4e11cc7bc4d2f922fbd8/common/common.c#L2068 `str_contains_token()` (or maybe exposing it in `libupsclient`, to avoid duplication?) so some contrived `ups.status="LOBOTOMIZED"` would not match `OB` as a sub-string.

Reply:
Should be addressed by https://github.com/networkupstools/nut/pull/2859 (to be available since NUT v2.8.3).

Reply:
Good point, mostly ported over the existing logic from NUT at that time (iirc), but definitely makes sense and nicely implemented in subsequent commits as usual. 👍